test/plugins/jsonBodyParser.test.js

```diff
@@ -490,3 +490,947 @@ describe('JSON body parser', function() {
         client.end();
     });
 });
+
+describe('JSON body parser with content type application/hal+json', function() {
```

Comment:
Wowza, this is a huge test! Thanks for adding all of these but I don't think we need to duplicate all of the existing tests. I'd just add a test or two to make sure that `*/*+json` gets matched as application/json.